tests/pds/pds_restore_test.go

```diff
@@ -1396,6 +1466,27 @@ var _ = Describe("{PerformRestoreAfterDataServiceUpdate}", func() {
 							Deployment: updatedDeployment,
 						}
 
+						stepLog = "Verify ds is tls enabled post ds update"
+						Step(stepLog, func() {
+							if ds.Name == mongodb && ds.DataServiceEnabledTLS {
+								connectionString, depPassword, port, err := pdslib.GetMongoDBConnectionString(updatedDeployment, ds.Name, namespace)
+								log.FailOnError(err, "error occured while getting connection string")
+
+								//Validate if TLS is enabled for the data service
+								err = controlPlane.ValidateIfTLSEnabled("pds", depPassword, connectionString, port)
+								dash.VerifyFatal(err.Error(), strings.Contains(err.Error(), ServerSelectionError) || strings.Contains(err.Error(), SocketError),
+									"error occured while validating if ds is TLS enabled")
+								//if err != nil {
```

Comment:
If the error is nil, signifying that a non-tls based connection was successful, This error is not captured